src/legacy/core_plugins/metrics/public/interpreter/index.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// @ts-ignore
+import { functionsRegistry } from 'plugins/interpreter/registries';
+
+/**
+ * Interpreter Service
+ */
+export class InterpreterService {
```

Comment:
Adding an interpreter service is not necessary here -- this will be part of the data plugin and passed in as a runtime dependency.

So to mirror the way it would work in new platform, you'll want to import the `functionsRegistry` and pass it to the plugin's setup method:

```typescript
import { Plugin } from './plugin';
// @ts-ignore
import { functionsRegistry } from 'plugins/interpreter/registries';

const core = {};
const plugins = {
  data: {
    expressions: {
      functionsRegistry
    },
  },
};

new Plugin().setup(core, plugins);
```

Once #39329 is merged, we should have a better idea of where exactly this will live in the data plugin. @streamich might have additional thoughts on this.

Reply:
agreed, we don't even need the interpreter, we just need the functionsRegistry

here is a sample PR (needs some updates, but you can see how functionregistry and visualizations get passed in to setup : https://github.com/elastic/kibana/pull/38767

Reply:
Nice catch! Changed.